src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,614 @@
+					else
+						StartingScrollIndicatorInsets = superScrollView.ScrollIndicatorInsets;
+				}
+			}
+		}
+
+		// If there was no LastScrollView, but there is a superScrollView,
+		// set the LastScrollView to be the superScrollView
+		else if (superScrollView is not null)
+		{
+			LastScrollView = superScrollView;
+			StartingContentInsets = superScrollView.ContentInset;
+			StartingContentOffset = superScrollView.ContentOffset;
+
+			if (OperatingSystem.IsIOSVersionAtLeast(11, 1))
+				StartingScrollIndicatorInsets = superScrollView.VerticalScrollIndicatorInsets;
+			else
+				StartingScrollIndicatorInsets = superScrollView.ScrollIndicatorInsets;
+		}
+
+		// Calculate the move for the ScrollViews
+		if (LastScrollView is not null)
+		{
+			var lastView = View;
+			superScrollView = LastScrollView;
+			nfloat innerScrollValue = 0;
+
+			while (superScrollView is not null)
+			{
+				var shouldContinue = false;
+
+				if (move > 0)
+					shouldContinue = move > -superScrollView.ContentOffset.Y - superScrollView.ContentInset.Top;
+
+				else if (superScrollView.FindResponder<UITableView>() is UITableView tableView)
+				{
+					shouldContinue = superScrollView.ContentOffset.Y > 0;
+
+					if (shouldContinue && View.FindResponder<UITableViewCell>() is UITableViewCell tableCell
+						&& tableView.IndexPathForCell(tableCell) is NSIndexPath indexPath
+						&& tableView.GetPreviousIndexPath(indexPath) is NSIndexPath previousIndexPath)
+					{
+						var previousCellRect = tableView.RectForRowAtIndexPath(previousIndexPath);
+						if (!previousCellRect.IsEmpty)
+						{
+							var previousCellRectInRootSuperview = tableView.ConvertRectToView(previousCellRect, RootController.Superview);
+							move = (nfloat)Math.Min(0, previousCellRectInRootSuperview.GetMaxY() - topLayoutGuide);
+						}
+					}
+				}
+
+				else if (superScrollView.FindResponder<UICollectionView>() is UICollectionView collectionView)
+				{
+					shouldContinue = superScrollView.ContentOffset.Y > 0;
+
+					if (shouldContinue && View.FindResponder<UICollectionViewCell>() is UICollectionViewCell collectionCell
+						&& collectionView.IndexPathForCell(collectionCell) is NSIndexPath indexPath
+						&& collectionView.GetPreviousIndexPath(indexPath) is NSIndexPath previousIndexPath
+						&& collectionView.GetLayoutAttributesForItem(previousIndexPath) is UICollectionViewLayoutAttributes attributes)
+					{
+						var previousCellRect = attributes.Frame;
+
+						if (!previousCellRect.IsEmpty)
+						{
+							var previousCellRectInRootSuperview = collectionView.ConvertRectToView(previousCellRect, RootController.Superview);
+							move = (nfloat)Math.Min(0, previousCellRectInRootSuperview.GetMaxY() - topLayoutGuide);
+						}
+					}
+				}
+
+				else
+				{
+					if (cursorRect.Y - innerScrollValue >= topLayoutGuide && cursorRect.Y - innerScrollValue <= keyboardYPosition)
+						shouldContinue = false;
+					else
+						shouldContinue = true;
```

Comment:
why not set the variable assignment with the if statement?
shouldContinue = !(cursorRect.Y - innerScrollValue >= topLayoutGuide && cursorRect.Y - innerScrollValue <= keyboardYPosition)

Reply:
Good call!